persia/env.py

```diff
@@ -1,156 +1,91 @@
-    RANK_ID = None
-    LOCAL_RANK = None
-
-    ENV_LAUNCHER = None
-
-    def env_check(self) -> bool:
-        """Check current environment is valid or not."""
-
-        rank_env = (
-            self.WORLD_SIZE is not None
-            and self.RANK_ID is not None
-            and self.LOCAL_RANK is not None
-        )
-        replica_env = self.REPLICA_INDEX is not None and self.REPLICA_SIZE is not None
-
-        if not (rank_env or replica_env):
-            raise Exception(
-                f"""PersiaEnv check failed, aleast pass rank or replica info to run the persia task,
-                    launcher: {self.ENV_LAUNCHER} world_size: {self.WORLD_SIZE},
-                    rank_id: {self.RANK_ID}, local_rank: {self.LOCAL_RANK}, replica_size: {self.REPLICA_SIZE},
-                    replica_index: {self.REPLICA_INDEX}"""
-            )
-
-    @cached_property
-    def rank(self):
-        # TODO: Check the environment and cache the value.
-        return self.RANK_ID
-
-    @cached_property
-    def local_rank(self):
-        return self.LOCAL_RANK
-
-    @cached_property
-    def replica_index(self):
-        return self.REPLICA_INDEX
-
-    @cached_property
-    def replica_size(self):
-        return self.REPLICA_SIZE
-
-    @cached_property
-    def world_size(self):
-        return self.WORLD_SIZE
-
-
-class HonchoENV(ENV):
-
-    ENV_LAUNCHER = "honcho"
-
+class _Env:
     def __init__(self):
-        """Honcho environment have the basic ability to support local training, typically used at single machine training."""
-        honcho_process_name = os.environ["HONCHO_PROCESS_NAME"]
-
-        if os.environ.get("RANK", None):
-            self.RANK_ID = int(os.environ["RANK"])
-            self.LOCAL_RANK = int(os.environ["LOCAL_RANK"])
-            self.WORLD_SIZE = int(os.environ.get("WORLD_SIZE"))
-        else:
-            _, process_idx = honcho_process_name.split(".")
-            process_idx = int(process_idx) - 1
-
-            self.REPLICA_INDEX = int(os.environ.get("REPLICA_INDEX"))
-            self.REPLICA_SIZE = int(os.environ.get("REPLICA_SIZE"))
+        self.replica_size = None
+        self.replica_index = None
+        self.world_size = None
+        self.rank = None
+        self.local_rank = None
+        self.is_init = False
 
+    def init(self):
+        if self.is_init:
+            return
 
-class DockerENV(ENV):
-
-    ENV_LAUNCHER = "docker"
-
-    def __init__(self):
-        """Docker environment that launch the PersiaML task by docker-compose."""
         if os.environ.get("RANK", None):
-            self.RANK_ID = int(os.environ["RANK"])
-            self.LOCAL_RANK = int(os.environ["LOCAL_RANK"])
-            self.WORLD_SIZE = int(os.environ.get("WORLD_SIZE"))
+            self.rank = int(os.environ["RANK"])
+            self.local_rank = int(os.environ["LOCAL_RANK"])
+            self.world_size = int(os.environ["WORLD_SIZE"])
+            assert self.rank >= 0, "RANK should not be a negative."
```

Comment:
"cannot be negative", similar for other places below

Reply:
Update at 6711266